File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/material/update/MaterialUpdate.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/update/IdListUpdateValue" ], function(stjs, IdListUpdateValue) {
	var MaterialUpdate = function() {
		this.properties = {};
		this.tagIds = new IdListUpdateValue();
	};
	stjs.extend(MaterialUpdate, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.material.update.MaterialUpdate';
		constructor.serialVersionUID = 1;

		prototype.materialId = null;
		prototype.properties = null;
		prototype.tagIds = null;

		prototype.getMaterialId = function() {
			return this.materialId;
		};
		prototype.setMaterialId = function(materialId) {
			this.materialId = materialId;
		};
		prototype.getProperty = function(propertyName) {
			return this.properties[propertyName];
		};
		prototype.setProperty = function(propertyName, propertyValue) {
			this.properties[propertyName] = propertyValue;
		};
		prototype.getProperties = function() {
			return this.properties;
		};
		prototype.setProperties = function(properties) {
			this.properties = properties;
		};
		prototype.getTagIds = function() {
			return this.tagIds;
		};
		prototype.setTagActions = function(actions) {
			this.tagIds.setActions(actions);
		};
	}, {
		materialId : "IMaterialId",
		properties : {
			name : "Map",
			arguments : [ null, null ]
		},
		tagIds : {
			name : "IdListUpdateValue",
			arguments : [ "ITagId" ]
		}
	});
	return MaterialUpdate;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/material/update/MaterialUpdate.java
/*
 * Copyright 2013 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.material.update;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IPropertiesHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IObjectUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IdListUpdateValue;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.ListUpdateValue.ListUpdateAction;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.material.id.IMaterialId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.tag.id.ITagId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Jakub Straszewski
 */
@JsonObject("as.dto.material.update.MaterialUpdate")
public class MaterialUpdate implements IUpdate, IObjectUpdate<IMaterialId>, IPropertiesHolder
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private IMaterialId materialId;

    @JsonProperty
    private IdListUpdateValue<ITagId> tagIds = new IdListUpdateValue<ITagId>();

    @JsonProperty
    private Map<String, String> properties = new HashMap<String, String>();

    @Override
    @JsonIgnore
    public IMaterialId getObjectId()
    {
        return getMaterialId();
    }

    @JsonIgnore
    public IMaterialId getMaterialId()
    {
        return materialId;
    }

    @JsonIgnore
    public void setMaterialId(IMaterialId materialId)
    {
        this.materialId = materialId;
    }

    @Override
    @JsonIgnore
    public void setProperty(String propertyName, String propertyValue)
    {
        properties.put(propertyName, propertyValue);
    }

    @Override
    @JsonIgnore
    public String getProperty(String propertyName)
    {
        return properties != null ? properties.get(propertyName) : null;
    }

    @Override
    @JsonIgnore
    public void setProperties(Map<String, String> properties)
    {
        this.properties = properties;
    }

    @Override
    @JsonIgnore
    public Map<String, String> getProperties()
    {
        return properties;
    }

    @JsonIgnore
    public IdListUpdateValue<ITagId> getTagIds()
    {
        return tagIds;
    }

    @JsonIgnore
    public void setTagActions(List<ListUpdateAction<ITagId>> actions)
    {
        tagIds.setActions(actions);
    }
}
